src/code42cli/cmds/securitydata.py

```diff
@@ -431,16 +492,36 @@ def send_to(
     HOSTNAME format: address:port where port is optional and defaults to 514.
     """
     cursor = _get_cursor(state, use_checkpoint)
-    handlers = ext.create_send_to_handlers(
-        state.sdk, FileEventExtractor, cursor, use_checkpoint, state.logger
-    )
-    _extract(
-        state, handlers, begin, end, or_query, advanced_query, saved_search, **kwargs
-    )
 
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        # if checkpoint name exists, checkpoint should be that eventId,
+        # otherwise it should set the initial value to ""
+        checkpoint = cursor.get(checkpoint_name) or ""
+
+        # older app versions stored checkpoint as float timestamp.
+        # we handle those here until the next run containing events will store checkpoint as the last eventId
+        if isinstance(checkpoint, (int, float)):
+            state.search_filters.append(InsertionTimestamp.on_or_after(checkpoint))
+            checkpoint = ""
+    else:
+        checkpoint = ""
+
+    query = _construct_query(state, begin, end, saved_search, advanced_query, or_query)
+    events = _get_all_file_events(state, query, checkpoint)
+
+    if not events:
+        click.echo("No results found.")
+        return
 
-def _get_file_event_extractor(sdk, handlers):
-    return FileEventExtractor(sdk, handlers)
+    for event in events:
+        if use_checkpoint:
+            checkpoint_name = use_checkpoint
+            cursor.replace(checkpoint_name, event["eventId"])
+        with warn_interrupt():
```

Comment:
The `warn_interrupt()` context should wrap the whole for loop, otherwise the user would need to hit ctrl-c twice within the timeframe it takes to process a single event, otherwise the counter would get reset.